### ISSUE TITLE
Binary is not compatible with Windows XP

### ISSUE THREAD

**Issue by bgK** [Doom64/fluidsynth-lite#5]
The following commit introduces the use of slim reader/writer locks:
https://github.com/Doom64/fluidsynth-lite/commit/5db9b88b4f08f58e886e3309776bfa4cbd719901

That synchronization primitive was not available until Windows Vista. At the ScummVM project, we use fluidsynth-lite for some platforms including Windows. However we still have some users that use Windows XP. Would it be possible to use another API for the mutex implementation?

**Comment by dotfloat** [comment]
Hey, sorry for responding so late.

Is this still an issue that needs a solution? I can add Win XP support, but might not be able to test it fully.

**Comment by bgK** [comment]
Thanks for your reply. It is indeed still an issue. I can test proposed fixes with Windows XP if needed.

**Comment by paulsapps** [comment]
you can probably restore the old locking defines with a WINVER check?

**Comment by dotfloat** [comment]
Yeah, so this is what I started working on and it got pretty messy so I put it on hold. Upstream fluidsynth uses GLib to provide the thread and atomic implementation, and what I did in this project is to replace that and have a bunch of `#ifdef`s for the popular implementations of these.

The correct way to do this would've been to make `fluid_thread_new` and co into their own functions, a `.c` for each platform, just like GLib does it.
